delatore/sources/influx.py

```diff
@@ -202,6 +205,113 @@ async def _get_status(self, metric):
+
+    _threshold = 60
+
+    @property
+    def threshold(self):
+        return self._threshold
+
+    @threshold.setter
+    def threshold(self, threshold):
+        self._threshold = threshold
+
+
+    async def get_update(self) -> dict:
+        self._metrics = self.params().metrics
+        last_elapsed_time, last_time_ms, status = await self._get_status(self._metrics[0])
+        self.list_elapsed_time.pop(0)
+        self.list_elapsed_time.append(last_elapsed_time)
+        results = []
+        if min(self.list_elapsed_time) < 60:
+            results.append(generate_status(self._metrics[0].name, status,
+                                           last_time_ms.strftime(UNIFIED_TIME_PATTERN)))
+        else:
+            results.append(generate_error('InfluxdbLBTiming', await self._get_error_message()))
+        return generate_message(self.CONFIG_ID, results)
+
+    async def _get_error_message(self):
+        with open(_CONFIGS['error_message_lb_timing.txt'], 'r') as file:
+            str_template = file.read()
+            list_queries = []
+        for met in self._metrics[1:]:
+            list_queries.append(met.query.format(entity=met.metric_id))
+        results_queries = await asyncio.gather(*[
+            self.influx_client.query(query) for query in list_queries
+        ])
+        results_queries_without_empty_lines = []
+        for result_query in results_queries:
+            try:
+                results_queries_without_empty_lines.append(result_query.raw['series'][0]['values'][0][1])
+            except KeyError:
+                results_queries_without_empty_lines.append('No data')
+        str_template = str_template.format(cpu_utilization=results_queries_without_empty_lines[0],
+                                           node_distribution=results_queries_without_empty_lines[1],
+                                           network_bytes_recv=results_queries_without_empty_lines[2],
+                                           network_bytes_send=results_queries_without_empty_lines[3])
+        return str_template
+
+    async def _get_status(self, metric):
+        query = metric.query.format(entity=metric.metric_id)
+        last_record = await self.influx_client.query(query)
+        try:
+            last_time = last_record.raw['series'][0]['values'][0][0]
+            last_elapsed_time = last_record.raw['series'][0]['values'][0][1]
+        except KeyError:
+            return generate_status(metric.name, Status.NO_DATA, None)
+        now = datetime.utcnow().timestamp()
+        last_time_ms = _convert_time(last_time)
+        status = Status.OK
+        if now - last_time_ms.timestamp() > metric.timeout:
+            status = Status.FAIL
+        return last_elapsed_time, last_time_ms, status
+
+
+class InfluxSourceLBDOWN(InfluxSource):
+    CONFIG_ID = 'influxdb_lb_down'
+    _params: InfluxParams = None
+    _error_message_file = 'error_message_lb_down.txt'
```

Comment:
Actual text messages  should be part of outputs, in sources you should use unified json for errors